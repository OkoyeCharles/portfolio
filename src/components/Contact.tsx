import { keyframes } from "@emotion/react";
import React from "react";
import { Reveal } from "react-awesome-reveal";
import ContactForm from "./ContactForm";


const slideUp = keyframes`
  from {
    opacity: 0;
    transform: translateY(2em);
  },
  to {
    opacity: 1;
    transform: translateY(0);
  }
`

interface ContactProps {
  testmode: boolean,
}

const Contact: React.FC<ContactProps> = ({ testmode }) => {
  return (
    <section className="contact">
      <a id="contact"></a>
      <div className="contact__container">
        <h2 className="contact__header">Contact</h2>
        <div className="contact__content">
          <Reveal triggerOnce keyframes={slideUp} duration={1500}>
            <p className="contact__coa">
              If you have an <strong>application</strong> you are interested in
              developing, a <strong>feature</strong> that you need built or a{" "}
              <strong>project</strong> that needs coding. I'd love to help with
              it.
            </p>
          </Reveal>
          {!testmode && (<ContactForm />)}
        </div>
      </div>
    </section>
  );
};

export default Contact;
