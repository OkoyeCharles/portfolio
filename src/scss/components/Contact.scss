@use "../abstracts" as *;

.contact {
  background: $dark;
}

.contact * {
  color: $light;
}

.contact__container {
  max-width: 1600px;
  margin-inline: auto;
  padding: 1em 1.8em 5em 4.5em;
}

.contact__header {
  width: fit-content;
  font-size: 1.5em;
  position: relative;
  text-transform: uppercase;
  margin-bottom: 2em;
}

.contact__header::after {
  position: absolute;
  content: "";
  bottom: -0.25em;
  left: 0.5rem;
  width: 100%;
  height: 0.75em;
  background: transparentize($color: $theme, $amount: 1 - 0.25);
}

.contact__content {
  max-width: 800px;
  margin-inline: auto;
  margin-block: 5em;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.contact__coa {
  text-align: center;
  margin-bottom: 2em;
  font-size: 1.375em;
  font-weight: 300;
  color: transparentize($color: $light, $amount: 1 - 0.75);
}

.contact__coa strong {
  color: transparentize($color: $light, $amount: 1 - 0.9);
}

@media screen and (max-width: 700px) {
  .contact__container {
    padding: 1.8em;
  }

  .contact__coa {
    font-size: clamp(1em, 3.5vw, 1.25em);
  }
}