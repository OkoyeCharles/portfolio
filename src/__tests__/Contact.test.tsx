import renderer from 'react-test-renderer';
import Contact from '../components/Contact';

it('<Contact /> component renders properly', () => {
  const component = renderer.create(
    <Contact testmode={true} />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});