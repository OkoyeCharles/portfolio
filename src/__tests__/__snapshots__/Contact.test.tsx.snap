// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> component renders properly 1`] = `
<section
  className="contact"
  id="contact"
>
  <div
    className="contact__container"
  >
    <h2
      className="contact__header"
    >
      Contact
    </h2>
    <div
      className="contact__content"
    >
      <div
        className="css-sanqhm"
        style={
          Object {
            "animationDelay": "0ms",
          }
        }
      >
        <p
          className="contact__coa"
          style={Object {}}
        >
          If you have an 
          <strong>
            application
          </strong>
           you are interested in developing, a 
          <strong>
            feature
          </strong>
           that you need built or a
           
          <strong>
            project
          </strong>
           that needs coding. I'd love to help with it.
        </p>
      </div>
    </div>
  </div>
</section>
`;
